File: core/src/main/java/io/github/fourlastor/game/demo/state/map/TileType.java
package io.github.fourlastor.game.demo.state.map;

import java.util.Arrays;

public enum TileType {
    TERRAIN("terrain", true),
    WATER("water", false),
    SOLID("solid", false),
    ;

    public final String mapName;
    public final boolean allowWalking;

    TileType(String mapName, boolean allowWalking) {
        this.mapName = mapName;
        this.allowWalking = allowWalking;
    }

    public static TileType fromMap(String mapName) {
        return Arrays.stream(values())
                .filter((it) -> it.mapName.equalsIgnoreCase(mapName))
                .findFirst()
                .get();
    }
}

File: core/src/main/java/io/github/fourlastor/game/demo/state/map/TileType.kt
package io.github.fourlastor.game.demo.state.map

enum class TileType(val mapName: String, val allowWalking: Boolean) {
  TERRAIN("terrain", true),
  WATER("water", false),
  SOLID("solid", false);

  companion object {
    fun fromMap(mapName: String): TileType =
        values().first { it.mapName.equals(mapName, ignoreCase = true) }
  }
}
